barretenberg/cpp/src/barretenberg/eccvm/eccvm_prover.cpp

```diff
@@ -276,7 +196,7 @@ template <IsECCVMFlavor Flavor> void ECCVMProver_<Flavor>::execute_transcript_co
 
 template <IsECCVMFlavor Flavor> HonkProof& ECCVMProver_<Flavor>::export_proof()
 {
-    proof = transcript->export_proof();
+    proof = transcript->proof_data;
```

Comment:
We apparently need to return proof_data here rather than `transcript->export_proof()` because the former sets `num_fr_written` to 0 and if we serialise and deserialise proofs than things get confused because that var is checked to establish whether there is actual stuff in transcript to retrieve. Maybe the API should be renamed/better documented, talking to Lucas about it.